src/pharmpy/model/statements.py

```diff
@@ -340,15 +340,34 @@ def move_dose(self, source, destination):
             Source compartment
         destination : Compartment
             Destination compartment
+        admid : int
+            Move dose with specified admid, move all if None. Default is None.
+        replace : bool
+            Replace dose of destination or add to. Default to True
 
         """
-        new_source = source.replace(dose=None)
-        new_dest = destination.replace(dose=source.dose)
+        if admid:
+            new_source_dose = tuple([d for d in source.doses if d.admid != admid])
+            new_dest_dose = tuple([d for d in source.doses if d.admid == admid])
+        else:
+            new_source_dose = tuple()
+            new_dest_dose = source.doses
+        new_source = source.replace(doses=new_source_dose)
+        if replace:
+            new_dest = destination.replace(doses=source.doses)
+        else:
+            new_dest = destination.replace(doses=destination.doses + new_dest_dose)
         mapping = {source: new_source, destination: new_dest}
         nx.relabel_nodes(self._g, mapping, copy=False)
+        return new_source, new_dest
+
+    def set_dose(self, compartment, dose, replace=True, admid=None):
+        """Set dose of compartment, replacing the previous if existing
```

Comment:
The first line of the docstring is misleading given that replacing is optional

Reply:
Changed to "Set dose of compartment, replacing the previous _by default_". Also added the `replace` and `admid` to "Parameters"